sympy/functions/combinatorial/numbers.py

```diff
@@ -1861,9 +1861,19 @@ def stirling(n, k, d=None, kind=2, signed=False):
 def _nT(n, k):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if k > n or k == 0:
+        return 0
+    if n == k or k == 1:
+        return 1
+    if 2*k >= n:
+        return partition._partition(n - k)
+    M = n
```

Comment:
No need for an else since the 3 cases above never get here.

Reply:
Somehow we need to reference that this nugget was modified from [here](https://github.com/klocey/partitions/blob/master/pypartitions/pypartitions.py).

Reply:
Rather than working with `n*k` we can work with (n-k)*k` with this, I believe:
```python
    M = n - k
    p = [1]*M
    for i in range(2, k + 1):
        for j in range(i + 1, M):
            p[j] += p[j - i]
        M -= 1
    # if p[0] were appended to the end of p then the last
    # k values of p are the nT(n, j) values for 0 < j < k in reverse order
    # p[-1] = nT(n, 1), p[-2] = nT(n, 2), etc.... Instead of putting the 1
    # from p[0] there, however, it is simply added to the sum below.
    return 1 + sum(p[1 - k:]) # ok for 1 < k <= n//2
```